extensions/data-plane/data-plane-api/src/main/java/org/eclipse/dataspaceconnector/dataplane/api/transfer/DataPlanePublicApiRequestFilter.java

```diff
@@ -119,6 +119,17 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
      */
     private DataFlowRequest createDataFlowRequest(ClaimToken claims, ContainerRequestContext requestContext) throws IOException {
         var dataAddress = typeManager.readValue(claims.getClaims().get(DATA_ADDRESS), DataAddress.class);
+
+        // extract the difference between base and request uri as dataaddress name -> httpdatasource
+        String baseUri = requestContext.getUriInfo().getBaseUri().toString();
+        String requestUri = requestContext.getUriInfo().getRequestUri().toString();
+        // TODO what about arbitrary deployment paths under the base path -> now become part of the name
+        String diffUri = requestUri.substring(baseUri.length());
+        if (diffUri.indexOf("?") >= 0) {
+            diffUri = diffUri.substring(0, diffUri.indexOf("?"));
+        }
```

Comment:
isn't that exactly what https://javadoc.io/static/jakarta.ws.rs/jakarta.ws.rs-api/3.0.0/jakarta/ws/rs/core/UriInfo.html#getPath() delivers - the path part of the URI relative to the base path?